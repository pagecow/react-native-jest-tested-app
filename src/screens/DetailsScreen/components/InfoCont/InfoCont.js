import React from 'react';
import { View, Text } from 'react-native';
import styles from './InfoCont.style';

export default function InfoCont({ header, text, type }) {
    if(header === "Phone") {
        text = text.replace(/\D+/g, '').replace(/(\d{3})(\d{3})(\d{4})/, '($1) $2-$3');
    }

    if(header === "Birthdate") {
        let months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"]
        let date = text.split('-');
        if(date[1] > 12 || date[1] < 1 || date[2] > 31 || date[2] < 1) {
            return null;
        } else {
            let month = months[Math.abs(date[1] - 1)]; 
            text = month + ` ${date[2]}, ${date[0]}`;
        }
    }

    if(header === "Address") {
       text = `${text.street}\n${text.city}, ${text.state} ${text.zipCode}, ${text.country}`
    }
    
    return (   
        <View style={styles.infoCont}>
            <Text style={styles.infoHeader}>{header}:</Text> 
            <View style={styles.infoTextCont}>
                <Text style={styles.infoText}>{text}</Text> 
                {type ? <Text style={styles.infoPhoneType}>{type}</Text> : null}
            </View>
        </View>
    );
  };