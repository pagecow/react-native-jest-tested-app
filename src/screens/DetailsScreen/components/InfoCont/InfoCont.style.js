import { StyleSheet } from 'react-native';

export default StyleSheet.create({
    infoCont: {
        width: "87%",
        marginLeft: "6.5%",
        justifyContent: "center",
        borderBottomWidth: 1.5,
        borderBottomColor: "#f1f1f1",
        marginBottom: 20,
    },
    infoHeader: {
        textTransform: 'uppercase',
        fontFamily: 'OpenSans_600SemiBold',
        color: "#8f8f8f80",
        marginBottom: 10,
    },
    infoTextCont: {
        flexDirection: "row",
        width: "100%",
        justifyContent: "space-between",
        marginBottom: 20,
    },
    infoText: {
        fontFamily: 'OpenSans_600SemiBold',
    },
    infoPhoneType: {
        fontFamily: 'OpenSans_600SemiBold',
        color: "#8f8f8f",
    }
});