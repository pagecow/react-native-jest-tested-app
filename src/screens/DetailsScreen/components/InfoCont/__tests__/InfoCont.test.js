import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import InfoCont from '../InfoCont';

describe('<InfoCont />', () => {
    it('renders without crashing', () => {
        const tree = renderer.create(<InfoCont />).toJSON();
        expect(tree).toMatchSnapshot();
    });


// test to see if the function will return null if the month or day is too great.
    it('renders null if Birthdate month number is over 12', () => {
        const comp = shallow(<InfoCont header="Birthdate" text="1975-13-31" />);
        expect(comp.get(0)).toBeFalsy();
    });

    it('renders null if Birthdate month number is under 1', () => {
        const comp = shallow(<InfoCont header="Birthdate" text="1975-0-31" />);
        expect(comp.get(0)).toBeFalsy();
    });

    it('renders null if Birthdate day number is over 31', () => {
        const comp = shallow(<InfoCont header="Birthdate" text="1975-12-32" />);
        expect(comp.get(0)).toBeFalsy();
    });

    it('renders null if Birthdate day number is under 1', () => {
        const comp = shallow(<InfoCont header="Birthdate" text="1975-12-0" />);
        expect(comp.get(0)).toBeFalsy();
    });

});



