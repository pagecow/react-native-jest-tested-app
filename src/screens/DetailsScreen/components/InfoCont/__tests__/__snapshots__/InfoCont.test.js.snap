// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoCont /> renders without crashing 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#f1f1f1",
      "borderBottomWidth": 1.5,
      "justifyContent": "center",
      "marginBottom": 20,
      "marginLeft": "6.5%",
      "width": "87%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#8f8f8f80",
        "fontFamily": "OpenSans_600SemiBold",
        "marginBottom": 10,
        "textTransform": "uppercase",
      }
    }
  >
    :
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "OpenSans_600SemiBold",
        }
      }
    />
  </View>
</View>
`;
